### PR TITLE
WEBAPP-441: Upgrade packages

### PR DESCRIPTION
**Pull request by Schedulaar**
I couldn't upgrade react-redux (https://issues.integreat-app.de/browse/WEBAPP-460).
I couldn't manage to fix tests for I18nProvider (https://issues.integreat-app.de/browse/WEBAPP-461).
I ignored some flow-files of redux-first-router, because they are broken with new flow-version. (https://issues.integreat-app.de/browse/WEBAPP-462)

This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/). 
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **WEBAPP**.

### CODE REVIEW

File: src/polyfills.js [hunk 1]
```diff
@@ -5,3 +5,5 @@ import '@babel/polyfill'
 import 'whatwg-fetch'
 
 import 'url-polyfill'
+
+window.regeneratorRuntime = require('regenerator-runtime') // todo: remove again
```
Comment (NKlug) [inline src/polyfills.js]: Should this be removed before merging?

File: src/setupTest.js [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import 'raf/polyfill'
 import { configure } from 'enzyme'
 import Adapter from 'enzyme-adapter-react-16'
+window.regeneratorRuntime = require('regenerator-runtime') // todo remove
```
Comment (NKlug) [inline src/setupTest.js]: Maybe also remove?

**Comment by Schedulaar** [comment]
How do we continue here @maxammann ?

File: src/setupTest.js (hunk 2 above)
Comment (Schedulaar) [reply src/setupTest.js]: Was already removed.

File: src/polyfills.js (hunk 1 above)
Comment (Schedulaar) [reply src/polyfills.js]: Was already removed

File: src/__mocks__/react-i18next.js [hunk 3]
```diff
@@ -4,15 +4,14 @@ import * as React from 'react'
 import i18next from 'i18next'
 
 // this mock makes sure any components using the translate HoC receive the t function as a prop
-export const withNamespaces = <Props> (namespace: string) => (Component: React.ComponentType<Props>) => {
-  const Translated = (props: Props) => <Component t={key => `${namespace}:${key}`} {...props} />
-  Translated.displayName = `Translate(${Component.displayName || Component.name})`
+const withTranslation = <Props> (namespace: string) => (Component: React.ComponentType<Props>) => {
+  const i18n = i18next.createInstance()
```
Comment (maxammann) [inline src/__mocks__/react-i18next.js]: Why do we need an i18next instance in here? I think it is only used in the tests and expected to be anything.

**Pull request merged**